s2/cellunion.go

```diff
@@ -570,7 +570,7 @@ func (cu *CellUnion) decode(d *decoder) {
 		return
 	}
 	n := d.readInt64()
-	if d.err != nil {
+	if d.err != nil || n < 0 {
```

Comment:
What do C++/Java do? Should this be read as uint64 / should the error be updated like below?

Reply:
cpp actually gets a uint_64t there, that's a better fix! https://github.com/google/s2geometry/blob/b99a7941052c1347ed84a508669d3155c2963b20/src/s2/s2cell_union.cc#L546